pkg/compact/downsample/downsample.go

```diff
@@ -148,57 +150,106 @@ func Downsample(
 
 		// Raw and already downsampled data need different processing.
 		if origMeta.Thanos.Downsample.Resolution == 0 {
+			var prevEnc chunkenc.Encoding = chks[0].Chunk.Encoding()
+
 			for _, c := range chks {
+				if prevEnc != c.Chunk.Encoding() {
```

Comment:
If previous encoding is float histogram and current encoding is normal histogram, can we downsample those chunks together since the input is same, float histogram?